snafu/stressng_wrapper/trigger_stressng.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import subprocess
+import time
+import logging
+import yaml
+
+logger = logging.getLogger("snafu")
+
+class Trigger_stressng():
+    def __init__(self, args):
+        self.uuid = args.uuid
+        self.runtype = args.runtype
+        self.timeout = args.timeout
+        self.vm_stressors = args.vm_stressors
+        self.vm_bytes = args.vm_bytes
+        self.mem_stressors = args.mem_stressors
+
+    def _run_stressng(self):
+        cmd = "stress-ng --job /workload/jobfile --log-file stressng.lg -Y stressng.yml"
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        return stdout.strip().decode("utf-8"), process.returncode
+
+    def _parse_outfile(self):
+        stream = open('stressng.yml', 'r')
+        data = yaml.load(stream, Loader=yaml.FullLoader)
+        metrics = (data["metrics"])
+        results = []
+        for metric in metrics:
+            stressor = (metric["stressor"])
+            bogoops = (metric["bogo-ops"])
+            result = {stressor : bogoops}
+            print(result)
+            results.append(result)
+        return results
+
+    def _json_payload(self, data, uuid, runtype, timeout, vm_stressors, vm_bytes, mem_stressors, timestamp):
+        logger.info("generating json payload")
+        edict = {}
+        processed = []
+        edict.update({
+            "workload": "stressng",
+            "uuid": uuid,
+            "runtype": runtype,
+            "timeout": timeout,
+            "vm_stressors": vm_stressors,
+            "vm_bytes": vm_bytes,
+            "mem_stressors": mem_stressors,
+            "timestamp": timestamp
+        })
+        for i in range(len(data)):
+            edict.update(dict(data[i]))
```

Comment:
You can combine two dictionaries with just 
```suggestion
        edict.update(data)`
```

Reply:
TIL...